packages/codegen/tests/types-builder.spec.ts

```diff
@@ -0,0 +1,357 @@
+import { expect, describe, it, beforeAll } from "vitest"
+import fs from "fs/promises"
+import assert from "assert"
+
+import {
+  defaultDeclarations,
+  getDocsTypesBuilder,
+  getTypesBuilder,
+} from "@/types-builder"
+import { MetadataLookup } from "@polkadot-api/metadata-builders"
+import { V14, V15, metadata } from "@polkadot-api/substrate-bindings"
+import { getLookupFn } from "@polkadot-api/metadata-builders"
+import knownTypes from "@/known-types"
+import { getChecksumBuilder } from "@polkadot-api/metadata-builders"
+import { getDispatchErrorId } from "@/generate-descriptors"
+
+describe("types-builder", () => {
+  let metadataLookup: MetadataLookup
+  let metadataDecoded: V14 | V15
+  let checksumBuilder: ReturnType<typeof getChecksumBuilder>
+
+  const getBuilder = () =>
+    getTypesBuilder(
+      defaultDeclarations(),
+      metadataLookup,
+      knownTypes,
+      checksumBuilder,
+    )
+
+  beforeAll(async () => {
+    const metadataRaw = await fs.readFile("./tests/ksm.bin")
```

Comment:
If possible use the ksm metadata we have already in `@polkadot-api/metadata-fixtures`, this way we don't have to duplicate a metadata file everywhere 🙃 

You can find an example in `packages/metadata-builders/tests/dynamic-builder.spec.ts`

Reply:
Sure, replaced it with a symlink.  
Was a bit hesitant to do that initially, as it's kind of an implicit coupling that tends to confusingly break when moving files around.